lib/Core/EpisodeAction/EpisodeActionSaver.php

```diff
@@ -33,25 +33,28 @@ public function __construct(
 	/**
 	 * @param array $episodeActionsArray
 	 *
-	 * @return void
+	 * @return EpisodeActionEntity[]
 	 */
-	public function saveEpisodeActions($episodeActionsArray, string $userId): void
+	public function saveEpisodeActions($episodeActionsArray, string $userId): array
 	{
 		$episodeActions = $this->episodeActionReader->fromArray($episodeActionsArray);
 
+		$episodeActionEntities = [];
+
         foreach ($episodeActions as $episodeAction) {
 			$episodeActionEntity = $this->hydrateEpisodeActionEntity($episodeAction, $userId);
 
 			try {
-                $this->episodeActionWriter->save($episodeActionEntity);
+                $episodeActionEntities[] = $this->episodeActionWriter->save($episodeActionEntity);
             } catch (UniqueConstraintViolationException $uniqueConstraintViolationException) {
-                $this->updateEpisodeAction($episodeActionEntity, $userId);
+                $episodeActionEntities[] = $this->updateEpisodeAction($episodeActionEntity, $userId);
             } catch (Exception $exception) {
                 if ($exception->getReason() === Exception::REASON_UNIQUE_CONSTRAINT_VIOLATION) {
-                    $this->updateEpisodeAction($episodeActionEntity, $userId);
+                    $episodeActionEntities[] = $this->updateEpisodeAction($episodeActionEntity, $userId);
                 }
             }
         }
+		return $episodeActionEntities;
```

Comment:
Now this function returns EpisodeActionEntity[] just for the tests. I'm not sure if this is the best way, but I can't think of another.